homeassistant/components/backup/manager.py

```diff
@@ -0,0 +1,173 @@
+"""Backup manager for the Backup integration."""
+from __future__ import annotations
+
+from dataclasses import asdict, dataclass
+import hashlib
+import json
+from pathlib import Path
+from tarfile import TarError
+from tempfile import TemporaryDirectory
+
+from securetar import SecureTarFile, atomic_contents_add
+
+from homeassistant.const import __version__ as HAVERSION
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.util import dt, json as json_util
+
+from .const import EXCLUDE_FROM_BACKUP, LOGGER
+
+
+@dataclass
+class Backup:
+    """Backup class."""
+
+    slug: str
+    name: str
+    date: str
+    path: Path
+    size: float
+
+    def as_dict(self) -> dict:
+        """Return a dict representation of this backup."""
+        return {**asdict(self), "path": self.path.as_posix()}
+
+
+class BackupManager:
+    """Backup manager for the Backup integration."""
+
+    _backups: dict[str, Backup] = {}
```

Comment:
We're setting a class attribute to a mutable object. Is that intended? Do we require all instances of this class to modify the same object? If that is not a requirement, eg we only expect one instance for the lifetime of Home Assistant, I'd recommend initializing the mutable object in the init method below.

Reply:
Only a single instance of this will exist.

Reply:
How would you recommend patching that in the tests?
Moving it breaks all tests that currently patches it, for the manager tests I can set it directly, but I'm not allowed to access hass.data in tests so I can not do that for the WebSocket tests.

Reply:
First I would decide at what level we should patch.

If we think it's ok to patch inside the integration we should patch the public methods of the backup manager, eg `BackupManager.remove_backup`, that are accessed by the websocket commands. Then it doesn't matter what the `_backups` attribute is. If we don't think it's ok to patch inside the integration we should patch the file and buffer library APIs, eg `securetar`, and not touch any integration details.

Reply:
https://github.com/home-assistant/core/pull/67431